packages/editor/src/lib/editor/managers/UserPreferencesManager.ts

```diff
@@ -26,15 +37,22 @@ export class UserPreferencesManager {
 			color: this.getColor(),
 			animationSpeed: this.getAnimationSpeed(),
 			isSnapMode: this.getIsSnapMode(),
+			colorScheme: this.user.userPreferences.get().colorScheme,
 			isDarkMode: this.getIsDarkMode(),
 			isWrapMode: this.getIsWrapMode(),
 		}
 	}
 	@computed getIsDarkMode() {
-		return (
-			this.user.userPreferences.get().isDarkMode ??
-			(this.inferDarkMode ? userPrefersDarkUI() : false)
-		)
+		switch (this.user.userPreferences.get().colorScheme) {
+			case 'dark':
+				return true
+			case 'light':
+				return false
+			case 'system':
+				return this.systemColorScheme.get() === 'dark'
+			default:
+				return this.inferDarkMode ? this.systemColorScheme.get() === 'dark' : false
```

Comment:
as mentioned elsewhere, `inferDarkMode` will only be respected when `colorScheme` is undefined, but at the moment that can't happen without the sdk user explicitly setting it to `undefined` because our default user preferences sets it to `'light'`